@claude please investigate why the unit tests have fail and propose a fix:
```
tests/unit/test_target_filtering.py::TestEndToEndTargetFilteringChain::test_workflow1_w1_1_receives_host_targets [06:46:23] ERROR    MainThread:Task                                 trace.py:309
                    secator.celery.run_command[884f4458-2f26-4c09-8             
                    05c-b9f09be528a1] raised unexpected:                        
                    EncodeError(PicklingError("Can't pickle <class              
                    'tests.unit.test_target_filtering.w2_2'>: it's              
                    not the same object as                                      
                    tests.unit.test_target_filtering.w2_2"))                    
                    ╭───── Traceback (most recent call last) ─────╮             
                    │ /opt/pipx/venvs/secator/lib/python3.12/site │             
                    │ -packages/kombu/serialization.py:41 in      │             
                    │ _reraise_errors                             │             
                    │                                             │             
                    │    38 def _reraise_errors(wrapper,          │             
                    │    39 │   │   │   │   │   include=(Exceptio │             
                    │    40 │   try:                              │             
                    │ ❱  41 │   │   yield                         │             
                    │    42 │   except exclude:                   │             
                    │    43 │   │   raise                         │             
                    │    44 │   except include as exc:            │             
                    │                                             │             
                    │ /opt/pipx/venvs/secator/lib/python3.12/site │             
                    │ -packages/kombu/serialization.py:220 in     │             
                    │ dumps                                       │             
                    │                                             │             
                    │   217 │   │   │   content_encoding = self._ │             
                    │   218 │   │                                 │             
                    │   219 │   │   with _reraise_errors(EncodeEr │             
                    │ ❱ 220 │   │   │   payload = encoder(data)   │             
                    │   221 │   │   return content_type, content_ │             
                    │   222 │                                     │             
                    │   223 │   def loads(self, data, content_typ │             
                    │                                             │             
                    │ /opt/pipx/venvs/secator/lib/python3.12/site │             
                    │ -packages/kombu/serialization.py:347 in     │             
                    │ pickle_dumps                                │             
                    │                                             │             
                    │   344 │   you to python clients.            │             
                    │   345 │   """                               │             
                    │   346 │   def pickle_dumps(obj, dumper=pick │             
                    │ ❱ 347 │   │   return dumper(obj, protocol=p │             
                    │   348 │                                     │             
                    │   349 │   registry.register('pickle', pickl │             
                    │   350 │   │   │   │   │     content_type='a │             
                    ╰─────────────────────────────────────────────╯             
                    PicklingError: Can't pickle <class                          
                    'tests.unit.test_target_filtering.w2_2'>: it's              
                    not the same object as                                      
                    tests.unit.test_target_filtering.w2_2 
```

--- a/tests/unit/test_target_filtering.py
+++ b/tests/unit/test_target_filtering.py
@@ -509,8 +509,14 @@ class TestEndToEndTargetFilteringChain(unittest.TestCase):
         After clear_modules(), the module-level w1_1/w2_1/etc. classes have stale bases
         (PythonRunner from the old import). Celery's pickle fails because Target objects
         created via those stale classes don't match sys.modules classes.
-        We rebuild fresh classes and update MOCK_TASK_CLASSES in-place.
+        We rebuild fresh classes, update MOCK_TASK_CLASSES in-place, and register the
+        new classes in this module's globals so pickle can resolve them by qualified name
+        (required when needs_chunking() triggers break_task which builds a chord with
+        mark_runner_completed.s(runner=task), causing the task instance to be pickled).
         """
+        import sys as _sys
+        _this_module = _sys.modules[__name__]
+
         from secator.runners import PythonRunner as _PythonRunner
         from secator.output_types import Port as _Port, Tag as _Tag, Url as _Url
         from secator.output_types import Technology as _Technology
@@ -583,6 +589,14 @@ class TestEndToEndTargetFilteringChain(unittest.TestCase):
 
         _w2_3.__name__ = 'w2_3'
         _w2_3.__qualname__ = 'w2_3'
+
+        # Register rebuilt classes in module globals so Celery pickle can find them by
+        # __module__.__qualname__ (tests.unit.test_target_filtering.w2_2, etc.)
+        _this_module.w1_1 = _w1_1
+        _this_module.w1_2 = _w1_2
+        _this_module.w2_1 = _w2_1
+        _this_module.w2_2 = _w2_2
+        _this_module.w2_3 = _w2_3
 
         # Update MOCK_TASK_CLASSES in-place so patched_discover_tasks returns fresh classes
         MOCK_TASK_CLASSES.clear()

